wat/WatLexer.g4

```diff
@@ -0,0 +1,280 @@
+
+NAT : Nat ;
+INT : Int ;
+FLOAT : Float ;
+STRING : String ;
+VALUE_TYPE : NXX ;
+CONST : NXX '.const' ;
+
+FUNCREF: 'funcref' ;
+MUT: 'mut' ;
+
+NOP: 'nop' ;
+UNREACHABLE: 'unreachable' ;
+DROP: 'drop' ;
+BLOCK: 'block' ;
+LOOP: 'loop' ;
+END: 'end' ;
+BR: 'br' ;
+BR_IF: 'br_if' ;
+BR_TABLE: 'br_table' ;
+RETURN: 'return' ;
+IF: 'if' ;
+THEN: 'then' ;
+ELSE: 'else' ;
+SELECT: 'select' ;
+CALL: 'call' ;
+CALL_INDIRECT: 'call_indirect' ;
+
+LOCAL_GET: 'local.get' ;
+LOCAL_SET: 'local.set' ;
+LOCAL_TEE: 'local.tee' ;
+GLOBAL_GET: 'global.get' ;
+GLOBAL_SET: 'global.set' ;
+
+LOAD : NXX '.load' (MEM_SIZE '_' SIGN)? ;
+STORE : NXX '.store' (MEM_SIZE)? ;
+
+OFFSET_EQ_NAT : 'offset=' Nat ;
+ALIGN_EQ_NAT : 'align=' Nat ;
+
+UNARY
+  : IXX '.clz'
+  | IXX '.ctz'
+  | IXX '.popcnt'
+  | FXX '.neg'
+  | FXX '.abs'
+  | FXX '.sqrt'
+  | FXX '.ceil'
+  | FXX '.floor'
+  | FXX '.trunc'
+  | FXX '.nearest'
+  ;
+
+BINARY
+  : IXX '.add'
+  | IXX '.sub'
+  | IXX '.mul'
+  | IXX '.div_s'
+  | IXX '.div_u'
+  | IXX '.rem_s'
+  | IXX '.rem_u'
+  | IXX '.and'
+  | IXX '.or'
+  | IXX '.xor'
+  | IXX '.shl'
+  | IXX '.shr_s'
+  | IXX '.shr_u'
+  | IXX '.rotl'
+  | IXX '.rotr'
+  | FXX '.add'
+  | FXX '.sub'
+  | FXX '.mul'
+  | FXX '.div'
+  | FXX '.min'
+  | FXX '.max'
+  | FXX '.copysign'
+  ;
+
+TEST
+  : IXX '.eqz'
+  ;
+
+COMPARE
+  : IXX '.eq'
+  | IXX '.ne'
+  | IXX '.lt_s'
+  | IXX '.lt_u'
+  | IXX '.le_s'
+  | IXX '.le_u'
+  | IXX '.gt_s'
+  | IXX '.gt_u'
+  | IXX '.ge_s'
+  | IXX '.ge_u'
+  | FXX '.eq'
+  | FXX '.ne'
+  | FXX '.lt'
+  | FXX '.le'
+  | FXX '.gt'
+  | FXX '.ge'
+  ;
+
+CONVERT
+  : 'i32.wrap_i64'
+  | 'i64.extend_i32_s'
+  | 'i64.extend_i32_u'
+  | 'f32.demote_f64'
+  | 'f64.promote_f32'
+  | IXX '.trunc_f32_s'
+  | IXX '.trunc_f32_u'
+  | IXX '.trunc_f64_s'
+  | IXX '.trunc_f64_u'
+  | FXX '.convert_i32_s'
+  | FXX '.convert_i32_u'
+  | FXX '.convert_i64_s'
+  | FXX '.convert_i64_u'
+  | 'f32.reinterpret_i32'
+  | 'f64.reinterpret_i64'
+  | 'i32.reinterpret_f32'
+  | 'i64.reinterpret_f64'
+  ;
+
+MEMORY_SIZE : 'memory.size' ;
+MEMORY_GROW : 'memory.grow' ;
+
+TYPE: 'type' ;
+FUNC: 'func' ;
+START: 'start' ;
+PARAM: 'param' ;
+RESULT: 'result' ;
+LOCAL: 'local' ;
+GLOBAL: 'global' ;
+TABLE: 'table' ;
+MEMORY: 'memory' ;
+ELEM: 'elem' ;
+DATA: 'data' ;
+OFFSET: 'offset' ;
+IMPORT: 'import' ;
+EXPORT: 'export' ;
+
+MODULE : 'module' ;
+BIN : 'binary' ;
+QUOTE : 'quote' ;
+
+SCRIPT: 'script' ;
+REGISTER: 'register' ;
+INVOKE: 'invoke' ;
+GET: 'get' ;
+ASSERT_MALFORMED: 'assert_malformed' ;
+ASSERT_INVALID: 'assert_invalid' ;
+ASSERT_UNLINKABLE: 'assert_unlinkable' ;
+ASSERT_RETURN: 'assert_return' ;
+ASSERT_RETURN_CANONICAL_NAN: 'assert_return_canonical_nan' ;
+ASSERT_RETURN_ARITHMETIC_NAN: 'assert_return_arithmetic_nan' ;
+ASSERT_TRAP: 'assert_trap' ;
+ASSERT_EXHAUSTION: 'assert_exhaustion' ;
+INPUT: 'input' ;
+OUTPUT: 'output' ;
+
+VAR : Name ;
+
+SPACE
+  : [ \t\r\n] -> skip
+  ;
+
+COMMENT
+  : ( '(;' .*? ';)'
+  | ';;' .*? '\n')-> skip
```

Comment:
Maybe it makes sense to put at least comments to the `HIDDEN` channel? But I would like to avoid the `skip` command at all.